src/odemis/driver/zeiss.py

```diff
@@ -438,28 +467,34 @@ class Stage(model.Actuator):
     def __init__(self, name, role, parent, rng=None, **kwargs):
         """
         inverted (set of str): names of the axes which are inverted
-        rng (dict str -> (float,float)): axis name -> min/max of the position on
-          this axis. Note: if the axis is inverted, the range passed will be
-          inverted. Also, if the hardware reports position outside of the range,
-          move might fail, as it is considered outside of the range.
+        rng (dict str -> (float,float)): axis name -> min/max of the position on this axis.
+        Note: if the axis is inverted, the range passed will be inverted.
+        Also, if the hardware reports position outside the range, move might fail, as it is considered outside the range.
         """
 
         if rng is None:
             rng = {}
 
+        # Ranges are from the documentation
         if "x" not in rng:
             rng["x"] = (5e-3, 152e-3)
         if "y" not in rng:
             rng["y"] = (5e-3, 152e-3)
         if "z" not in rng:
             rng["z"] = (5e-3, 40e-3)
-
-        axes_def = {
-            # Ranges are from the documentation
-            "x": model.Axis(unit="m", range=(rng["x"][0], rng["x"][1])),
-            "y": model.Axis(unit="m", range=(rng["y"][0], rng["y"][1])),
-            "z": model.Axis(unit="m", range=(rng["z"][0], rng["z"][1])),
-        }
+        if "rx" not in rng:
+            rng["rx"] = (0, 0)
+        if "rz" not in rng:
+            rng["rz"] = (0, 0)
+        if "m" not in rng:
+            rng["m"] = (0, 0)
+
+        axes_def = {}
+        for ax, r in rng.items():
+            rng_ax = (r[0], r[1])  # Force a tuple, and check the user passed 2 values
+            if rng_ax != (0, 0):  # if range is not specified exclude the axis
```

Comment:
That behaviour sounds good to me, *except* that `(0, 0)` is not really nice to indicate the axis isn't supported. Please, replace by the special value `None` (which will be written `null` in YAML).
Also, indicate this behaviour in the docstring about `rng`.